image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,205 @@
+  cluster infrastructure platform status.
+- The registry will bootstrap itself as `Removed` on the following platform types:
+  - `BareMetal` (i.e. baremetal platforms that support IPI install)
+  - `oVirt` (RHV)
+  - `VSphere`
+  - `None` (true UPI baremetal installs)
+- The registry will bootstrap itself as `Managed` with `EmptyDir` storage on
+  the following platform types:
+  - `Libvirt`
+  - `Unknown` platforms
+- The registry will continue to bootstrap itself as `Managed` with specific
+  storage on the following platforms:
+  - `AWS` (Amazon S3)
+  - `Azure` (Azure blob storage)
+  - `GCP` (Google Cloud Storage)
+  - `OpenStack` (Swift)
+- Any platform that wishes to set up the registry must perform a day 2 action
+  that does the following:
+  - Provisions suitable storage, typically a PV that supports the `RWX` access
+    mode.
+  - Updates the image registry operator's management state to `Managed`.
+- Telemeter should be aware if a cluster's registry has been removed:
+  - Reason codes in the ClusterOperator conditions (`Available` in particular).
+  - Prometheus alerts warning the registry has been removed.
+
+### Risks and Mitigations
+
+#### Image registry is never installed
+
+Several components will likely report errors if the registry is `Removed` and
+an admin does nothing to set up the registry. Potential candidates that will
+break:
+
+1. Builds which push to an imagestream tag will error at runtime.
+2. Samples operator imagestreams which import from `registry.redhat.io` will
+   require a pull secret to be usable.
+3. Imagestreams imported with `--reference-policy=local` will not reference the
+   internal registry as expected. `BuildConfig` and `DeploymentConfig` objects
+   which do their own resolution of `ImageStreamTag` will break.
+
+Mitigations:
+
+1. Existing error messages from affected components.
+2. Image registry's ClusterOperator will provide `Reason="Removed"` if the
+   registry remains removed.
+3. If removed, the image registry operator should fire an alert warning that:
+   1. The registry has been removed.
+   2. The following components may not work as expected:
+      1. `ImageStreamTags`
+      2. `BuildConfigs` which reference `ImageStreamTags`
+      3. `DeploymentConfigs` which reference `ImageStreamTags`
+   3. Recommends admins configure storage and update the operator config to
+   the `Managed` state.
+
+## Design Details
+
+### Test Plan
+
+- Existing e2e-operator suite will need to verify that the registry behaves as
+  expeced when `Removed`:
+  - No registry components are installed.
+  - Reporting "green"/does not block upgrade.
+  - Fire an alert
+  - Registry should behave as expected when state is switched to `Managed`
+- RHV should have their own e2e/conformance suite to ensure the registry comes
+  up (likely out of scope for this enhancement)
+- UPI vSphere test suites will need to update the registry's state to `Managed`
+  in addition to configuring storage.
```

Comment:
Need to update the CI templates for platforms where the registry comes up "removed" to have those templates configure the registry to managed/emptydir in order to avoid failing CI tests due to no registry.

Reply:
@bparees noted impacts to `BareMetal` and `None` templates.

Reply:
why is vSphere any different from BareMetal and None? (and RHV for that matter).

i'd expect all 4 to have the same treatment from a CI/testing perspective.